### PR TITLE
Newsletters: add issue 55 (2019-07-17)

### PR DESCRIPTION
**Pull request by harding**
TODO:

- [x] Alternatively use a bech32 sending support section being composed by @moneyball (he's currently waiting on a source, so the section included in the first draft of this PR is a backup)
- [x] Possibly remove the link to transactionfee.info.  That site is currently incorrectly reporting legacy transactions as being segwit transactions.  I pinged the site author 0xB10C about the issue via Twitter DM a couple hours ago but haven't heard back yet (I think it's the normal time for sleeping where he lives, so this is expected).  If it's not fixed by publication time, I think we should drop the link so as not to confuse readers.
- [x] Add the rest of the newsletter when it's written.  :-)

### CODE REVIEW

File: _includes/specials/bech32/18-real-fees.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+seen to date were around 0.00001000 BTC/vbyte in December 2017 and January 2018.
+For that range, the following charts show the amount of money that may be saved for
+users of two common transaction templates, single-sig and 2-of-3
+multisig:
+
+![FIXME](/img/posts/2019-07-real-cost-p2pkh-p2wpkh.png)
+
+![FIXME](/img/posts/2019-07-real-cost-p2sh-p2wsh.png)
+
+For users of legacy transactions with other other transaction templates,
```
Comment (0xB10C) [inline _includes/specials/bech32/18-real-fees.md]: s/other other/other/

**Comment by 0xB10C** [comment]
Thank you harding for the head's up.  [transactionfee.info](https://transactionfee.info)  reported every transaction as SegWit due to a to me unknown change in an external API. This is fixed now.

fyi: [transactionfee.info](https://transactionfee.info) only shows savings for legacy -> nested SegWit. [blockstream.info](https://blockstream.info) shows nested and bech32 savings.

**Comment by harding** [comment]
Pushed two commits.  The first fixes the typo found by @0xB10C (thanks!) and drops TransactionFee.info because it doesn't show native segwit savings as described in the post (thanks for noting that, 0xB10C!).  The second commit makes some edits, adds this week's only news item, and adds another item to the notable commits section.

File: _includes/specials/bech32/18-real-fees.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+you may be able to learn roughly how much you will save in percentage
+terms by pasting in the txid of a typical transaction you sent into
+information sites such as an Esplora
+block explorer like [Blockstream.info][].  You can multiply that
+percentage by the vbyte size of your transaction to see how many vbytes
+you'd save.  Note that using a third-party service reveals to them that
+you have an interest in that transaction, possibly significantly
+reducing your privacy.  You can privately get the approximate vbyte
+savings by aggregating the byte size of all scriptSigs in your typical
+transaction and then multiplying that number by 0.75.  When you know how
```
Comment (jnewbery) [inline _includes/specials/bech32/18-real-fees.md]: I think everything from "You can privately..." onward could be made clearer by turning it into step-by-step instructions, ie:

1. Use a tool to parse the transaction. Bitcoin Core comes bundled with a tool called `bitcoin-tx` that can do this for you. Run `bitcoin-tx -json <hex_serialized_tx>`
2. Add up the total size of all scriptSigs. `bitcoin-tx` outputs the hex of each scriptSig. Halve the length of this hex string to get the size of the scriptSig.
3. Multiply the size of scriptSigs that value by 0.75 to get vbytes saved.
4. Multiply the vbytes saved by feerate to get the amount of bitcoin saved.
5. Multiply the amount of bitcoin saved by the exchange rate to get savings.

File: _posts/en/newsletters/2019-07-17-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+- [Bitcoin Core #15277][] allows the Linux versions of Bitcoin Core to
+  be deterministically compiled using [GNU Guix][] (pronounced "geeks").
+  This requires much less setup than the currently-still-supported
+  mechanism based on [Gitian][] and so it is hoped that it will lead to a
+  greater number of users being able to independently verify that the
+  release binaries are derived solely from the Bitcoin Core source code
+  and its dependencies.  Additionally, Guix requires fewer build
+  environment dependencies and there is ongoing work to essentially
+  eliminate its need for any pre-compiled binaries in the typical build
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-07-17-newsletter.md]: I'd replace 'essentially eliminate' with 'minimize'. Even with a hex0 bootstrap (https://www.gnu.org/software/mes/manual/mes.html#The-Magical-Self_002dHosting-Hex-Assembler), you'd still need to start with a minimal binary.

File: _posts/en/newsletters/2019-07-17-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  toolchain, both of which make the build system much easier to audit.
+  Altogether this should reduce the amount of trust users need to place
+  in the developers of Bitcoin Core and the software used to build
+  Bitcoin Core.  Although this merged PR only builds Linux binaries,
+  it is expected that support for Windows and macOS will
+  follow.
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-07-17-newsletter.md]: Perhaps link to Carl's talk at Breaking Bitcoin (https://www.youtube.com/watch?v=DKOG0BQMmmg&feature=youtu.be&t=19828) for people who want to learn more about bootstrappable and deterministic builds.

File: _posts/en/newsletters/2019-07-17-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+- [BOLTs #607][] extends the LN specification to allow packets to contain
+  records that start with a *tag* identifying their purpose, followed by
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-07-17-newsletter.md]: s/tag/type/ ?

I've checked the PR and BOLT1, and I don't see the word 'tag' being used.

File: _includes/specials/bech32/18-real-fees.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+transaction fees are paid in bitcoin terms rather than dollar terms.
+That means the real cost to send a transaction is expected to increase
+in correspondence with bitcoin prices even if feerates stay the same.
+
+We've previously discussed [how much users and services can save][bech32 savings
+segment] by switching to native segwit (bech32) addresses, but we've
+only described that in terms of vbyte and percentage savings.  In this
+bech32 sending support section, we look at the savings in real terms.
+
+The lowest practical fee is 0.00000001 BTC/vbyte.  The highest typical fees
```
[the remark this answers is not in the document]
Comment (harding) [reply _includes/specials/bech32/18-real-fees.md]: Other options:

- 1e-8 BTC/vbyte
- Color fade:

File: _posts/en/newsletters/2019-07-17-newsletter.md (hunk 3 above)
Comment (harding) [reply _posts/en/newsletters/2019-07-17-newsletter.md]: I chose my words carefully here---that hex0 binary isn't a compiler artifact, it's handwritten assembly that simply has its comments dropped and its hex transliterated into binary.  Therefore it seems true to me that they "essentially eliminate its need for any pre-compiled binaries".  However, if you still think that's misleading, I'm happy to rewrite.

File: _posts/en/newsletters/2019-07-17-newsletter.md (hunk 5 above)
Comment (harding) [reply _posts/en/newsletters/2019-07-17-newsletter.md]: /me :man_facepalming:

File: _includes/specials/bech32/18-real-fees.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+As of this writing, the price of bitcoin in national currencies has been
+rapidly increasing over the past few months.  We don't know whether this
+trend will continue (and it may reverse), but significant increases in
```
[the remark this answers is not in the document]
Comment (harding) [reply _includes/specials/bech32/18-real-fees.md]: I'd like to leave this in.  I agree it should be obvious but, without this caveat, I feel like we're price pumping---which is something I think a technical newsletter should avoid.

**Comment by harding** [comment]
Pushed two commits.  The first addresses feedback from @jnewbery (thanks!) minus that which I left comments on above.  The second replaces the bech32 section with the excellent one written by @moneyball.  I left the previous section in the PR and did not rename it so that all comments remain valid; I suggest we try to resolve any open threads on it, merge it, and then I'll rename it and use it in next week's PR.

File: _includes/specials/bech32/18-real-fees.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (0xB10C) [reply _includes/specials/bech32/18-real-fees.md]: I'm curious why the `sat/vByte` unit is problematic. Most people seem to understand it and it is the shortest. 

With the `BTC/vbyte` unit a [thin space](https://en.wikipedia.org/wiki/Thin_space) might even be (somewhat) better for readability than a full space:
-  0.000 000 01 BTC/vByte |-- full space
-  0.000&thinsp;000&thinsp;01 BTC/vByte |-- thin space (`&thinsp;`)
-  0.000<sub>&thinsp;</sub>000<sub>&thinsp;</sub>01 BTC/vByte |-- thin space & subscript (`0.000<sub>&thinsp;</sub>000<sub>&thinsp;</sub>01 BTC/vByte`)

[the remark this answers is not in the document]
Comment (jnewbery) [reply _includes/specials/bech32/18-real-fees.md]: We've discussed this internally at Optech a few times. I think Dave's main opposition to using 'satoshi' is similar to Greg's here: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-March/016709.html (that Satoshi didn't name the smallest subdivision after himself and that doing so is a bit cultish), but Dave, correct me if I'm wrong.

Like I said, we don't need to solve it this week, and we can discuss in https://github.com/bitcoinops/bitcoinops.github.io/issues/165.

File: _posts/en/newsletters/2019-07-17-newsletter.md (hunk 3 above)
Comment (jnewbery) [reply _posts/en/newsletters/2019-07-17-newsletter.md]: Sounds good to me!

File: _includes/specials/bech32/18-real-fees.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (harding) [reply _includes/specials/bech32/18-real-fees.md]: Good compromise!  Pushed a commit that makes this edit, thanks.

**Comment by jnewbery** [comment]
ACK 7336b37eaf06a30aa017f9fedce2ec51f77dafde

Thanks @harding !

File: _includes/specials/bech32/18-real-fees.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (bitschmidty) [reply _includes/specials/bech32/18-real-fees.md]: Marking this conversation as resolved. Note that this segment was bumped in favor of the bech32 support issues segment this week.

File: _includes/specials/bech32/18-real-fees.md (hunk 2 above)
Comment (bitschmidty) [reply _includes/specials/bech32/18-real-fees.md]: I like the step by step instructions. (note again, for a future newsletter)

**Comment by bitschmidty** [comment]
@harding comes through, never late
even when outside the lower 48

@moneyball explains what services do
when deciding to receive to bech32

@jnewbery with another review
and @0xB10C with one too

(thanks all!)

**Pull request merged**